lua/flatten/core.lua

```diff
@@ -138,37 +138,34 @@ M.edit_files = function(opts)
 			end
 			name = newname
 		end
+		files[#files + 1] = name
 		vim.api.nvim_buf_set_name(stdin_buf, name)
 	end
 
+	local winnr
+	local bufnr
+
 	-- Open window
 	if type(open) == "function" then
-		-- Pass list of new buffer IDs
-		local bufs = vim.api.nvim_list_bufs()
-		local start = #bufs - #files
 		-- Add buffer for stdin
 		local newbufs = {}
 		-- If there's an stdin buf, push it to the table
 		if stdin_buf then
-			start = start - 1
 			table.insert(newbufs, stdin_buf)
```

Comment:
The user won't have a way to know which buffer is the stdin buf as the name is based on the first line - maybe the stdin bufnr should be passed to open as a 3rd parameter?

Also, I don't think we need `newbufs` at all anymore, 151-155 should be able to be removed.

Reply:
Good idea, I removed `stdin_buf` from `files` and added it as the third argument of the `open` function.

Also removed lines 151-155.

Reply:
Perfect, thanks!